src/ngx_http_datadog_module.cpp

```diff
@@ -394,24 +482,36 @@ static ngx_int_t datadog_master_process_post_config(
   // configuration.  This means that the nginx configuration did not use the
   // `datadog` directive, and did not use any overridden directives, such as
   // `proxy_pass`.
-  const auto main_conf = static_cast<datadog_main_conf_t *>(
+  auto *const main_conf = static_cast<datadog_main_conf_t *>(
       ngx_http_cycle_get_module_main_conf(cycle, ngx_http_datadog_module));
 
   if (main_conf == nullptr) {
     // no config, no behavior
     return NGX_OK;
   }
 
+#ifdef WITH_WAF
+  ngx_http_next_output_body_filter = ngx_http_top_body_filter;
+  ngx_http_top_body_filter = output_body_filter;
+#endif
+
   // Forward tracer-specific environment variables to worker processes.
-  std::string name;
-  for (const auto &env_var_name :
-       TracingLibrary::environment_variable_names()) {
-    name = env_var_name;
-    if (const char *value = std::getenv(name.c_str())) {
-      main_conf->environment_variables.push_back(
-          environment_variable_t{.name = name, .value = value});
+  auto push_to_main_conf = [main_conf](const std::string &env_var_name) {
+    if (const char *value = std::getenv(env_var_name.c_str())) {
+      main_conf->environment_variables.push_back(environment_variable_t{
+          .name = std::move(env_var_name), .value = value});
```

Comment:
The copy of the std::string is usually not needed (only if the env variable is set), which is why it's passed by reference. But yes, the `std::move` is redundant for when the copy is needed (it will call the copy constructor in both cases).

Reply:
On the other hand you're creating a copy of the original `string_view` into a temporary, passing it as a const reference and then copying it again when the environment variable exists. A concern could be around the nul-termination of those views, as they need to be passed to `std::getenv`, however in both cases (tracing and appsec), they are originally literals. Either way, I suppose this is a safer approach if you believe the source of those names could eventually lack nul-termination, otherwise it might make more sense to just pass a `string_view`.

Reply:
Ah I should've checked the caller. You're right, it makes more sense to take an `std::string`.